internal/pkg/cli/pipeline_init.go

```diff
@@ -225,6 +242,51 @@ func (o *initPipelineOpts) validateURL(url string) error {
 	return nil
 }
 
+func (o *initPipelineOpts) validateURLExists(url string) error {
+	repos, err := o.fetchAndParseURLs()
+	if err != nil {
+		return fmt.Errorf("fetch and parse URLs: %w", err)
+	}
+	if _, ok := repos[url]; !ok {
+		return fmt.Errorf("URL '%s' is not a local git remote; please check that you're in the correct directory", o.repoURL)
+	}
+	o.repoShortName = repos[url]
+	return nil
+}
+
+func (o *initPipelineOpts) validateBranch() error {
```

Comment:
Also this is only for validating local branches (it could be not existed remotely)

Reply:
I'm not sure I understand. The `-a` flag has `git branch` return both local and remote branches. 
We're listing all branches that have `[remoteName]/*`.